### PR TITLE
NovelRT::Maths operator update - GeoVector<T> and GeoMatrix4x4<T>.

### PR DESCRIPTION
**Pull request by RubyNova**
This adds missing maths and boolean operators onto both of the specified types.

### CODE REVIEW

File: include/NovelRT.h [hunk 1]
```diff
@@ -150,7 +150,9 @@ namespace NovelRT::Windowing {
 
 #include "NovelRT/Animation/AnimatorPlayState.h"
 #include "NovelRT/Animation/SpriteAnimatorFrame.h"
-#include "NovelRT/Maths/GeoVector.h"
+#include "NovelRT/Maths/GeoVector2.h"
+#include "NovelRT/Maths/GeoVector3.h"
+#include "NovelRT/Maths/GeoVector4.h"
 #include "NovelRT/Maths/GeoMatrix4.h"
```
Comment (tannergooding) [inline include/NovelRT.h]: nit: should probably be `GeoMatrix4x4` in order to differentiate from `Matrix4x3` or other types that sometimes crop up.

File: include/NovelRT/Maths/GeoVector2.h [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  class GeoVector2 {
+    friend class Graphics::RenderObject;
+    friend class Input::InteractionService;
+    friend class GeoBounds;
+    template<typename U>
+    friend class GeoMatrix4;
+
+  private:
+    GeoVector2(glm::vec<2, T> value) : _value(glm::vec<4, T>(value, 0, 0)) {}
+    GeoVector2(glm::vec<4, T> value) : _value(value) {}
```
Comment (tannergooding) [inline include/NovelRT/Maths/GeoVector2.h]: Now that we have `GeoVector4`, this type should likely explicitly wrap a `glm::vec<2, T>` and we should have a `GeoVector4(GeoVector2 value, T z = 0, T w = 0)` overload (and similar for `GeoVector3`) instead.

It allows space savings, simplifies various bits, and makes the upcast to `vec<4, T>` explicit.

File: include/NovelRT/Maths/GeoVector2.h [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+#endif
+
+namespace NovelRT::Maths {
+  template<typename T>
+  class GeoVector2 {
+    friend class Graphics::RenderObject;
+    friend class Input::InteractionService;
+    friend class GeoBounds;
+    template<typename U>
+    friend class GeoMatrix4;
```
Comment (tannergooding) [inline include/NovelRT/Maths/GeoVector2.h]: I don't think `GeoVector2` or `GeoVector3` need to be friend with `GeoMatrix4`?

File: include/NovelRT/Maths/GeoVector3.h [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+#define NOVELRT_MATHS_GEOVECTOR3_H
+
+#ifndef NOVELRT_H
+#error Please do not include this directly. Use the centralised header (NovelRT.h) instead!
+#endif
+
+namespace NovelRT::Maths {
+  template<typename T>
+  class GeoVector3 {
+    friend class Graphics::RenderObject;
```
Comment (tannergooding) [inline include/NovelRT/Maths/GeoVector3.h]: Are all of these friends valid?

**Comment by tannergooding** [comment]
Overall LGTM. Just a few nits/questions.

File: include/NovelRT/Maths/GeoVector3.h (hunk 4 above)
Comment (RubyNova) [reply include/NovelRT/Maths/GeoVector3.h]: Cleaned up all friend classes for the vector types to only what's being used.

**Pull request merged**